proj/src/main/webapp/script.js

```diff
@@ -183,6 +192,148 @@ firebase.auth().onAuthStateChanged(firebaseUser => {
     }
 });
 
+
+/* Set by authState listener    */
+var allTagsRef;
+var abridgedTagsRef;
+var tagRemovalRef;
+var abridgedTagRemovalRef;
```

Comment:
same comment here about global variables. I think having everything defined within the scopes they are used is safer so that if different parts of the page are loading at different times, there is no chance of them modifying the same reference.